### PR TITLE
Enable cross build against scala 2.12 and 2.13

### LINKED ISSUE #136: Scala 2.13 Support

### PR DESCRIPTION
**Pull request by aumann**
Fixes #136

Adds htt4s-0-21 (http4s-0-20 only built upto 2.13_M5)

### CODE REVIEW

File: sbt-scalapact-nodeps/src/main/scala/com/itv/scalapact/plugin/ScalaPactPlugin.scala [hunk 1]
```diff
@@ -10,6 +10,8 @@ import sbt.plugins.JvmPlugin
 import sbt.{Def, _}
 import complete.DefaultParsers._
 
+import scala.language.implicitConversions
```
Comment (aumann) [inline sbt-scalapact-nodeps/src/main/scala/com/itv/scalapact/plugin/ScalaPactPlugin.scala]: That's the only 'real' change here. The rest has is scalafmt

File: build.sbt [hunk 2]
```diff
@@ -331,7 +394,7 @@ lazy val plugin =
     .dependsOn(pluginShared)
     .dependsOn(argonaut62)
     .dependsOn(http4s018)
-    .settings(compilerOptions212: _*)
+    .settings(scalacOptions ++= compilerOptionsVersion(scalaVersion.value))
```
Comment (dantb) [inline build.sbt]: Ditto but I guess this wants `scala212OnlySettings`?

File: build.sbt [hunk 3]
```diff
@@ -376,7 +440,7 @@ lazy val standalone =
     .dependsOn(core)
     .dependsOn(circe09)
     .dependsOn(http4s018)
-    .settings(compilerOptions212: _*)
+    .settings(scalacOptions ++= compilerOptionsVersion(scalaVersion.value))
```
Comment (dantb) [inline build.sbt]: Ditto

**Comment by aumann** [comment]
Sorry for the silence, will tackle the remaining issues today.

File: scalapact-http4s-0-20/src/main/scala/com/itv/scalapact/http4s20/impl/Http4sClientHelper.scala [hunk 4]
```diff
@@ -23,13 +23,17 @@ object Http4sClientHelper {
...
-      .withSslContextOption(sslContext)
       .withUserAgentOption(Option(`User-Agent`(AgentProduct("scala-pact", Option(BuildInfo.version)))))
       .withCheckEndpointAuthentication(false)
-      .resource
+    sslContext.fold(
+      builder.withDefaultSslContext
+    )(ctx =>
+      builder.withSslContext(ctx)
+    ).resource
+    builder.resource
```
[the remark this answers is not in the document]
Comment (aumann) [reply scalapact-http4s-0-20/src/main/scala/com/itv/scalapact/http4s20/impl/Http4sClientHelper.scala]: oh, yes. Didn't know about the SSLContext.getDefault -- much better

File: build.sbt [hunk 5]
```diff
@@ -317,7 +380,7 @@ lazy val pluginShared =
       scalaVersion := scala212
     )
     .dependsOn(core)
-    .settings(compilerOptions212: _*)
+    .settings(scalacOptions ++= compilerOptionsVersion(scalaVersion.value))
```
[the remark this answers is not in the document]
Comment (aumann) [reply build.sbt]: I changed these for uniformity, but left them at 2.12 -- to be honest I am not sure about cross compilation in the sbt plugin stuff

**Comment by aumann** [comment]
I addressed the comments w.r.t. the builders / build.sbt

Concerning the WartRemover.Any stuff:

- I tried fixing the http4s response warnings
- However, afterwards there were warnings everywhere in com.itv.scalapact.ScalaPactVerify
- Maybe this is related to https://github.com/wartremover/wartremover/issues/454

**Comment by davesmith00000** [comment]
Hi @aumann - sorry - our turn to be slow! Just wanted to say that this hasn't been forgotten, I'll try and take a look in the next couple of days.

**Comment by vilunov** [comment]
@dantb @davesmith00000 Hi! Is there any progress on reviewing this PR?

**Comment by orium** [comment]
Thanks @aumann.  I went ahead and solved the conflicts with master and did some minor tweaking.

**Pull request merged**